### ISSUE TITLE
airsenal_setup_initial_db --clean results in HTTPError: 404

### ISSUE THREAD

**Issue by markajam** [alan-turing-institute/AIrsenal#619]
I am very new to github. I hope this complies with contribution guidelines. Not sure how to label this?

I missed creating my team until after start of season (wk#1). I have now done so manually via FPL website. I have used airsenal last season.

I was having an initial problem which the fix from #617 was applied and appears to have solved.

Then after issuing  
`airsenal_setup_initial_db --clean`

I end up with the following.

...
Couldn't find player Ivan Cavaleiro
100%|█████████████████████████████████████████████████████████████████████████| 1107/1107 [00:01<00:00, 668.80it/s]
Getting initially selected players in squad _ID deleted_ for first gameweek...
Got starting squad from gameweek 2. Adding player data...
Traceback (most recent call last):
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/framework/data_fetcher.py", line 511, in _get_request
    r.raise_for_status()
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://fantasy.premierleague.com/api/entry/_ID deleted_/event/2/picks/

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/mark/miniconda3/envs/airsenalenv/bin/airsenal_setup_initial_db", line 8, in <module>
    sys.exit(main())
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/scripts/fill_db_init.py", line 90, in main
    make_init_db(args.fpl_team_id, seasons, dbsession)
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/scripts/fill_db_init.py", line 45, in make_init_db
    fill_initial_squad(fpl_team_id=fpl_team_id, dbsession=dbsession)
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/framework/transaction_utils.py", line 157, in fill_initial_squad
    init_players = get_players_for_gameweek(starting_gw, fpl_team_id)
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/framework/utils.py", line 892, in get_players_for_gameweek
    player_data = apifetcher.get_fpl_team_data(gameweek, fpl_team_id)["picks"]
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/framework/data_fetcher.py", line 246, in get_fpl_team_data
    fpl_team_data = self._get_request(
  File "/home/mark/miniconda3/envs/airsenalenv/lib/python3.9/site-packages/airsenal/framework/data_fetcher.py", line 513, in _get_request
    raise requests.HTTPError(f"{err_msg}: {e}") from e
requests.exceptions.HTTPError: Unable to access FPL team API https://fantasy.premierleague.com/api/entry/_ID deleted_/event/2/picks/: 404 Client Error: Not Found for url: https://fantasy.premierleague.com/api/entry/_ID deleted_/event/2/picks/
(airsenalenv) mark@STUDY:~/AIrsenal$

**Comment by jack89roberts** [comment]
Hi @markajam thanks for reporting the bug. I think it should be fixed now if you reinstall from the develop branch (where we work on te next AIrsenal release) as follows:

```
git pull
git checkout develop
conda activate airsenalenv
pip install -e .
airsenal_setup_initial_db --clean
```

But let us know if not (I've not tested it myself as I don't have a team entered GW2 to run it on!)

**Comment by Tdarnell** [comment]
Still problematic when first creating the DB, I think because some fixtures have null GW 

```
Traceback (most recent call last):
  File "/mnt/d/Code/AI/AIrsenal/.venv/bin/airsenal_setup_initial_db", line 5, in <module>
    from airsenal.scripts.fill_db_init import main
  File "/mnt/d/Code/AI/AIrsenal/airsenal/scripts/fill_db_init.py", line 9, in <module>
    from airsenal.framework.transaction_utils import fill_initial_squad
  File "/mnt/d/Code/AI/AIrsenal/airsenal/framework/transaction_utils.py", line 9, in <module>
    from airsenal.framework.utils import (
  File "/mnt/d/Code/AI/AIrsenal/airsenal/framework/utils.py", line 208, in <module>
    NEXT_GAMEWEEK = get_next_gameweek()
  File "/mnt/d/Code/AI/AIrsenal/airsenal/framework/utils.py", line 62, in get_next_gameweek
    earliest_future_gameweek = get_max_gameweek(season, dbsession) + 1
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int
```

![image](https://github.com/alan-turing-institute/AIrsenal/assets/4771442/f1950467-d21e-4436-b04f-4452e82dab24)

**Comment by jack89roberts** [comment]
Is this when using postgresql @Tdarnell ? I don't think it's related to this original issue but thanks for the fix - I'll comment on the PR.

**Comment by Tdarnell** [comment]
> Is this when using postgresql @Tdarnell ? I don't think it's related to this original issue but thanks for the fix - I'll comment on the PR.

Yes this is using postgres, I can open a new issue if it's unrelated

**Comment by jack89roberts** [comment]
I've merged your fix for it anyway so don't worry about the issue 🙂

**Comment by markajam** [comment]
> Hi @markajam thanks for reporting the bug. I think it should be fixed now if you reinstall from the develop branch (where we work on te next AIrsenal release) as follows:
> 
> ```
> git pull
> git checkout develop
> conda activate airsenalenv
> pip install -e .
> airsenal_setup_initial_db --clean
> ```
> 
> But let us know if not (I've not tested it myself as I don't have a team entered GW2 to run it on!)

Hi @jack89roberts This has solved the issue with 1.7.3. No errors and I can run as for last season all necessary steps. Thank you very much. Much to learn ... I shall leave you to close in case you need to add anything further.
